### ISSUE TITLE
H.C.B.B.S results and base location markers [S7] [Spoilers]

### ISSUE THREAD

**Issue by WeirdAlex03** [Dinip/hermitcraft#53]
**This contains spoilers about what "HCBBS" stands for and what it causes. If you care about it, go watch any hermit's video on it.**

------------------

So, thanks to the HermitCraft Big Base Swap, almost everyone's base is now somebody else's base. The swaps are as follows ([from u/McElligot on the HC subreddit](https://www.reddit.com/r/HermitCraft/comments/lj57g0/spoilers_megathread_2_for_the_hcbbs/gn94pio?utm_source=share&utm_medium=web2x&context=3)): 
> MumboJumbo <-> GoodTimesWithScar
> Docm77 <-> Cubfan135
> JoeHillsTSD <-> xBCrafted
> FalseSymmetry <-> TangoTek
> ZombieCleo <-> Keralis
> VintageBeef <-> Iskall85
> Grian <-> Zedaph
> StressMonster101 <-> ImpulseSV
> Maybe Rendog <-> iJevin (see original post)

This now raises the question of what we are going to do on the map. I assume the first thing is to wait a few episodes to see exactly how this ends up playing out but for now, are there any ideas on what we might do to indicate the original/current ownership?

**Comment by WeirdAlex03** [comment]
Ok since it seems no one else has anything to say, I'll make an update. It seems most if not all hermits intend to stay in their new bases for at least a little while. Ren's latest episode ended with the HCBBS event and Jevin hasn't been on HC in a month, so we'll have to wait at least another week to see if they decide to follow through.

For the rest of the Hermits that swapped, I'd change the icon and name updated to the new owner of the base/location. I don't know if it is possible to add an extra field to the map, if so then an optional one to show the original owner. Otherwise, that can be noted in parentheses at the end of the name (`B's base (originally A's)`").

I think my idea is good, but I'm not you, so I'm open to feedback/suggestions/criticism/really any interaction so I know I'm talking to a wall

**Comment by Dinip** [comment]
Hi,
As I said before, I'm currently not following much of season 7 episodes, so I'm a little lost about this topic... I trust that you will do the best to be accurate and keep the map updated!

> I don't know if it is possible to add an extra field to the map, if so then an optional one to show the original owner.

I think it is but I also like the idea of being something like you said `B's base (originally A's)`. I think it's easier to keep the marker's file as simple as possible but if you prefer that you, let me know 😀

**Comment by WeirdAlex03** [comment]
> As I said before, I'm currently not following much of season 7 episodes, so I'm a little lost about this topic... 

Tl;dr a strange object ~~that totally wasn't built by Mumbo~~ appeared in the shopping district advertising "change is coming" and "HCBBS". The hermits that put their name in had their base swapped with someone else who put their name in. For example, Mumbo's base now belongs to Scar and Scar's base now belongs to Mumbo.

> I think it is but I also like the idea of being something like you said `B's base (originally A's)`. I think it's easier to keep the marker's file as simple as possible but if you prefer that you, let me know 😀

That's a good point. I think it would be cleaner on the map with a separate field but it would make things a little more complicated on the backend to do that. I don't think it would be that much more complicated, but I'm familiar and comfortable with this stuff, someone unfamiliar should still be able to contribute and that could be off-putting.

I'll get started with the parentheses method and keep it standardized so it'll be easier to transition if there is follow-through on that front.

**Comment by Dinip** [comment]
> I don't think it would be that much more complicated, but I'm familiar and comfortable with this stuff, someone unfamiliar should still be able to contribute and that could be off-putting.

Yeah, for people that are familiar/used to json format (which, I guess, is not that complicated to understand) it would be fine. I'm not the developer of the map but I've done some work ([on my fork](https://github.com/Dinip/mapcrafter)) to fit my needs over the time (using mapcrafter for ~2 years) and even to fit Hermitcraft's map (some of the changes are not published since they are just specific to the case) and for me it would be fine do to add a new "key" to the marker object and try to fit that into marker body.

> I'll get started with the parentheses method and keep it standardized so it'll be easier to transition if there is follow-through on that front.

I think it is the best (at least for now), if sometime in the future, the change makes more sense, we (I) can adapt it with the data that is already present, to fit a new format.
